sktime/transformations/series/tests/test_lag.py

```diff
@@ -42,6 +42,27 @@ def test_lag_fit_transform_out_index(X, index_out):
     elif index_out == "shift":
         assert len(Xt) == len(X)
         assert X.index[2:].isin(Xt.index).all()
+        assert not X.index[:2].isin(Xt.index).any()
+
+
+@pytest.mark.parametrize("X", X_fixtures)
+@pytest.mark.parametrize("index_out", index_outs)
+def test_lag_fit_transform_out_values(X, index_out):
+    """Test that index sets of fit_transform output behave as expected."""
+    t = Lag(2, index_out=index_out)
+    if isinstance(X, pd.Series):
+        X = pd.DataFrame(X)
+    X_fit = X[:2]
+    X_trafo = X[2:]
+    Xt = t.fit(X_fit).transform(X_trafo)
+
+    if index_out in ["original", "extend"]:
+        assert all(Xt.iloc[0].values == X_fit.iloc[0].values)
```

Comment:
An alternative here would be to use:

```python
np.testing.assert_equal(Xt.iloc[2].values, X_fit.iloc[0].values)
```

or an equivalent of pandas instead of the assert statement. This should have the advantage that `np.array([np.nan])`  would be asserted to be equal with `np.array([np.nan])`, while `assert (np.array([np.nan]) == np.array([np.nan])).all()` would raise an exception.

Reply:
good point - is that just removing `values`?

Reply:
No, it is replacing line 60 with the proposed line of code. (If I got the question right)

Reply:
Oh, I see., `np.testing`. Makes sense.

I meant, do you know what is the "equivalent of pandas"?

Reply:
changed to `np.testing`

Reply:
pd.testing.assert_frame_equal or pd.testing.assert_series_equal is the corresponding pandas testing function. However, If I rethink it, this would probably not work since we change the index when applying the Lag. Thus, it is probably better to stay with numpy.testing here.